values/encoding/int_encode.go

```diff
@@ -96,7 +96,7 @@ func (enc *intEncoder) Encode(
 	}
 
 	if len(dictionary) <= maxCardinalityAllowed {
-		bytesPerDictionaryValue := int64(math.Ceil(float64(bits.Len(uint(valueMeta.Max-valueMeta.Min))) / 8.0))
+		bytesPerDictionaryValue := int64(math.Max(math.Ceil(float64(bits.Len(uint(valueMeta.Max-valueMeta.Min)))/8.0), 1.0))
```

Comment:
without this, if the range is 0, `bytesPerDictValue` is 0 and dictionary decoding will fail. Somehow all the dictionary encoding tests passed due to some miracle.

Also, could we encode this with bits instead of bytes?